### PR TITLE
`azurerm_mssql_elasticpool` : fix `license_type` can't be set to null when changing from vCore to DTU model issue

### PR DESCRIPTION
**Pull request by sinbai**
Per the following behavior of the [API](https://github.com/Azure/azure-rest-api-specs/blob/32c178f2467f792a322f56174be244135d2c907f/specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/ElasticPools.json#L118C10-L118C10), removed the check of TF. Also set a default value when the API does not return a value for license type property to fix issue #22311.

The behavior of the API:
image

Test results:
image

### CODE REVIEW

**Review by stephybun — CHANGES_REQUESTED** [review]
@sinbai per your table the behaviour is that the default value `LicenseIncluded` isn't returned for the older skus, in which case we should be defaulting it in the read instead of removing `Computed` from the property.

```
licenseType := string(sql.LicenseIncluded)
if string(properties.LicenseType) != "" {
  licenseType = string(properties.LicenseType)
}
d.Set("license_type", licenseType)
```

Could you please make that change?

**Comment by sinbai** [comment]
> @sinbai per your table the behaviour is that the default value `LicenseIncluded` isn't returned for the older skus, in which case we should be defaulting it in the read instead of removing `Computed` from the property.
> 
> ```
> licenseType := string(sql.LicenseIncluded)
> if string(properties.LicenseType) != "" {
>   licenseType = string(properties.LicenseType)
> }
> d.Set("license_type", licenseType)
> ```
> 
> Could you please make that change?

@stephybun thanks for your feedback. Sure. However, after updating code as above and removing the `Computed` from the property and deleting the ignore_changes of the `license_type` property, the test case `TestAccMsSqlElasticPool_vCoreToStandardDTU` fails with the following error:
image

I assume that for the older skus, the API does not return the default value `LicenseIncluded` because the API doesn't support specifying it. “Optional + Computed” means we can't set ` license_type` to null. If we don't remove the `Computed` from the property, which would cause the problem of issue #22311. WDYT?  Please feel free to let me know your suggestions.

**Comment by sinbai** [comment]
> @sinbai that error message is coming from the provider, but your table specifies that `license_type` can be set for all skus so this check can actually be removed?

Thanks for your suggestion. I have updated the code.  All test cases passed in TeamCity. Could you please take another look?

File: internal/services/mssql/mssql_elasticpool_resource_test.go [hunk 1]
```diff
@@ -268,6 +268,42 @@ func TestAccMsSqlElasticPool_hyperScale(t *testing.T) {
 	})
 }
 
+func TestAccMsSqlElasticPool_vCoreToStandardDTU(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mssql_elasticpool", "test")
+	r := MsSqlElasticPoolResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.no_licenseType(data),
```
Comment (stephybun) [inline internal/services/mssql/mssql_elasticpool_resource_test.go:277]: We don't use underscores in the test config names, can you please remove it and use camelCase like we do everywhere else in the provider 
```suggestion
			Config: r.noLicenseType(data),
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @sinbai. Test looks good, one minor thing needs to be fixed but otherwise this should be good to go.

File: internal/services/mssql/mssql_elasticpool_resource_test.go (hunk 1 above)
Comment (sinbai) [reply internal/services/mssql/mssql_elasticpool_resource_test.go]: Fixed.

**Comment by sinbai** [comment]
> Thanks @sinbai. Test looks good, one minor thing needs to be fixed but otherwise this should be good to go.

Thank you @stephybun, the name has been updated. Could you please take another look?

**Review by stephybun — APPROVED** [review]
Thanks @sinbai LGTM 🌴

**Pull request merged**